core/bn.js

```diff
@@ -88,12 +88,15 @@ sjcl.bn.prototype = {
   /**
    * Convert to a hex string.
    */
-  toString: function() {
-    this.fullReduce();
-    var out="", i, s, l = this.limbs;
-    for (i=0; i < this.limbs.length; i++) {
+  toString: function(doTrim) {
+    var i, s, l, out = new sjcl.bn(this);
+    out.fullReduce();
```

Comment:
`fullReduce` seems to be a stub and only calls `this.normalize()`. Actually a think that there's a lot of refactoring and optimization potential in the whole bn code...